server/devpi_server/model.py

```diff
@@ -338,17 +338,17 @@ def create_stage(self, index, type="stage", **kwargs):
                 "indexname '%s' contains characters that aren't allowed. "
                 "Any ascii symbol besides -.@_ is blocked." % index)
         kwargs.setdefault("volatile", True)
+        if type == "stage":
+            kwargs.setdefault("acl_upload", [self.name])
+            kwargs.setdefault("acl_toxresult_upload", [":ANONYMOUS:"])
+            kwargs.setdefault("bases", [])
+            kwargs.setdefault("mirror_whitelist", [])
         ixconfig = get_indexconfig(
             self.xom.config.hook,
             type=type, **kwargs)
-        if type == "stage":
-            if ixconfig.get("acl_upload") is None:
-                ixconfig["acl_upload"] = [self.name]
-            if ixconfig.get("acl_toxresult_upload") is None:
-                ixconfig["acl_toxresult_upload"] = [":ANONYMOUS:"]
+        if "bases" in ixconfig:
```

Comment:
might need to be moved to the ``if type == "stage"`` above

Reply:
It is gone  now with "Split index type specific parts of get_indexconfig into separate methods."